nessie/main_test.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/spf13/viper"
+	genclient "github.com/treeverse/lakefs/api/gen/client"
+	"github.com/treeverse/lakefs/api/gen/client/setup"
+	"github.com/treeverse/lakefs/api/gen/models"
+	"github.com/treeverse/lakefs/logging"
+)
+
+var (
+	logger logging.Logger
+	client *genclient.Lakefs
+	svc    *s3.S3
+)
+
+func TestMain(m *testing.M) {
+	systemTests := flag.Bool("system-tests", false, "Run system tests")
+	flag.Parse()
+	if !*systemTests {
+		os.Exit(0)
+	}
+	logger = logging.Default()
+
+	viper.SetDefault("setup_lakefs", true)
+	viper.SetDefault("setup_lakefs_timeout", 5*time.Minute)
+	viper.SetDefault("endpoint_url", "http://localhost:8000")
+	viper.SetDefault("s3_endpoint", "s3.local.lakefs.io:8000")
+	viper.SetDefault("access_key_id", "")
+	viper.SetDefault("secret_access_key", "")
+	viper.SetDefault("storage_namespace", "s3://nessie-system-testing")
+
+	viper.AddConfigPath(".")
+	viper.SetEnvPrefix("NESSIE")
+	viper.SetConfigName("nessie")
+	viper.AutomaticEnv()
+
+	err := viper.ReadInConfig()
+	if err != nil && !errors.As(err, &viper.ConfigFileNotFoundError{}) {
+		logger.Fatal("Failed to read configuration", err)
+	}
+
+	ctx := context.Background()
+
+	// initialize the env/repo
+	logger = logging.Default()
+	logger.WithField("settings", viper.AllSettings()).Info("Starting nessie")
+
+	endpointURL := viper.GetString("endpoint_url")
+	u, err := url.Parse(endpointURL)
+	if err != nil {
+		logger.Fatal("Failed to parse endpoint URL", endpointURL, err)
+	}
+
+	apiBasePath := genclient.DefaultBasePath
+	if u.Path != "" {
+		apiBasePath = u.Path
+	}
+	r := httptransport.New(u.Host, apiBasePath, []string{u.Scheme})
+	client = genclient.New(r, strfmt.Default)
+	if err := waitUntilLakeFSRunning(ctx, client); err != nil {
+		logger.Fatal("Waiting for lakeFS", err)
+	}
+
+	setupLakeFS := viper.GetBool("setup_lakefs")
+	if setupLakeFS {
+		// first setup of lakeFS
+		const adminUserName = "nessie"
+		res, err := client.Setup.SetupLakeFS(&setup.SetupLakeFSParams{
+			User: &models.Setup{
+				DisplayName: swag.String(adminUserName),
+			},
+			Context: ctx,
+		})
+		if err != nil {
+			logger.Fatal("Failed to setup lakeFS", err)
+		}
+
+		logger.WithFields(logging.Fields{
+			"access_key_id":     res.Payload.AccessKeyID,
+			"access_secret_key": res.Payload.AccessSecretKey,
+		}).Info("Cluster setup successfully")
+		viper.Set("access_key_id", res.Payload.AccessKeyID)
+		viper.Set("secret_access_key", res.Payload.AccessSecretKey)
+	}
+	r.DefaultAuthentication = httptransport.BasicAuth(viper.GetString("access_key_id"), viper.GetString("secret_access_key"))
+
+	s3Endpoint := viper.GetString("s3_endpoint")
+	awsSession := session.Must(session.NewSession())
+	svc = s3.New(awsSession,
+		aws.NewConfig().
+			WithRegion("us-east-1").
+			WithEndpoint(s3Endpoint).
+			WithDisableSSL(true).
+			WithCredentials(credentials.NewCredentials(
+				&credentials.StaticProvider{
+					Value: credentials.Value{
+						AccessKeyID:     viper.GetString("access_key_id"),
```

Comment:
You can't use the same credentials to authenticate to S3 and authenticate to lakeFS.

Reply:
not the same these are nessie's key/secret to lakefs